src/Layers/xrRender/r__dsgraph_build.cpp

```diff
@@ -55,10 +55,16 @@ void D3DXRenderBase::r_dsgraph_insert_dynamic(IRenderable* root, dxRender_Visual
     // a) Allow to optimize RT order
     // b) Should be rendered to special distort buffer in another pass
     VERIFY(pVisual->shader._get());
-    ShaderElement* sh_d = &*pVisual->shader->E[4]; // 4=L_special
-    if (RImplementation.o.distortion && sh_d && sh_d->flags.bDistort && pmask[sh_d->flags.iPriority / 2])
+    ref_selement element = pVisual->shader->E[4];
+    ShaderElement* sh_d;
+
+    if (element != nullptr)
```

Comment:
When comparing with constants it is better to set the constant first, this will help to experience less pain in the future.

Reply:
Also, this and other similar places need comment of reason of this checks (clang stuff)

Reply:
Pointers don't have to be checked against nullptr at all:
```
if (element)
...
```

Reply:
@nitrocaster
Yep, agree with you. I will change it.

Reply:
@eagleivg @nitrocaster 
Done.